### PR TITLE
Promote ReplicaSet Replace and Patch Test +2 Endpoints

### LINKED ISSUE #99134: Write ReplicaSet Replace and Patch Test +2 Endpoints
# Progress <code>[4/5]</code>

-   [X] APISnoop org-flow : [ReplicaSetScaleTest.org](https://github.com/cncf/apisnoop/blob/master/tickets/k8s/)
-   [X] test approval issue : [kubernetes/kubernetes#99134](https://github.com/kubernetes/kubernetes/issues/99134)
-   [x] test pr : kuberenetes/kubernetes[#99380](https://github.com/kubernetes/kubernetes/pull/99380)
-   [x] two weeks soak start date : [Link](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=ReplicaSet%20Replace%20and%20Patch%20tests)
-   [ ] two weeks soak end date : 8 March 2021
-   [ ] test promotion pr : kubernetes/kubernetes#?

# Identifying an untested feature Using APISnoop

According to this APIsnoop query, there are still some remaining Replicaset endpoints that are untested.

with this query, you can filter untested endpoints by their category and eligibility for conformance. e.g below shows a query to find all conformance eligible untested, stable, core endpoints

```sql-mode
SELECT
  endpoint,
  -- k8s_action,
  -- path,
  -- description,
  kind
  FROM testing.untested_stable_endpoint
  where eligible is true
  and endpoint like '%ReplicaSet%'
    and endpoint not like '%Status%'
  --and category = 'core'
  order by kind, endpoint desc
  limit 25;
```

```example
                  endpoint                  |    kind
--------------------------------------------|------------
 patchAppsV1NamespacedReplicaSet            | ReplicaSet
 listAppsV1ReplicaSetForAllNamespaces       | ReplicaSet
 deleteAppsV1CollectionNamespacedReplicaSet | ReplicaSet
(3 rows)

```
replaceAppsV1NamespacedReplicaSet  does not show up as untested as it touched by another [e2e test](https://github.com/ii/kubernetes/commit/d4b38853a591880c69c283268e580ccb46247ee5) not promoted to Conformance.

# API Reference and feature documentation

-   [Kubernetes API Reference Docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#-strong-write-operations-replicaset-v1-apps-strong-)
-   [client-go - ReplicaSet](https://pkg.go.dev/k8s.io/kubernetes/test/e2e/framework/replicaset)

# The mock test

## Test outline

1.  Create a ReplicaSet with a static label

2.  Read the ReplicaSet info

3.  Scale the ReplicaSets to 2

4.  Patch the ReplicaSet

5.  Read the ReplicaSet to ensure it is Patch

6.  Delete the Namespace and ReplicaSet

### Test the functionality in Go

Due to the complexity of setting up the resources for APIService we have used the current e2e test as template. It has been extended in a [new ginkgo test](https://github.com/ii/kubernetes/commit/d4b38853a591880c69c283268e580ccb46247ee5) for review.

# Verifying increase in coverage with APISnoop

```sql-mode
select distinct  endpoint, useragent
                 -- to_char(to_timestamp(release_date::bigint), ' HH:MI') as time
from testing.audit_event
where endpoint ilike '%ReplicaSet%'
and useragent ilike '%ReplicaSet lifecycle tests%'
-- and release_date::BIGINT > round(((EXTRACT(EPOCH FROM NOW()))::numeric)*1000,0) - 60000
-- and useragent like 'kubectl%'
order by endpoint
limit 100;

```

```example
               endpoint               |                                              useragent
--------------------------------------|------------------------------------------------------------------------------------------------------
 createAppsV1NamespacedReplicaSet     | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
 patchAppsV1NamespacedReplicaSet      | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
 readAppsV1NamespacedReplicaSet       | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
 replaceAppsV1NamespacedReplicaSet    | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
(4 rows)

```

# Final notes

If a test with these calls gets merged, ****test coverage will go up by 2 points****

This test is also created with the goal of conformance promotion.

---

/sig testing
/sig architecture
/area conformance
/sig apps

### PR DESCRIPTION
**Pull request by riaankleinhans**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
- replaceAppsV1NamespacedReplicaSet
- patchAppsV1NamespacedReplicaSet

**Which issue(s) this PR fixes:**
Fixes #99134

**Testgrid Link:**
[Link](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=ReplicaSet%20Replace%20and%20Patch%20tests)

**Special notes for your reviewer:**
Adds +2 endpoint test coverage (good for conformance)

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```

/sig testing
/sig architecture
/sig apps
/area conformance

### CODE REVIEW

File: test/conformance/testdata/conformance.yaml [hunk 1]
```diff
@@ -704,6 +704,13 @@
     the Job MUST execute to completion.
   release: v1.16
   file: test/e2e/apps/job.go
+- testname: ReplicaSet, is created, Replaced and Patched
+  codename: '[sig-apps] ReplicaSet Replace and Patch tests [Conformance]'
+  description: Create a ReplicaSet (RS) with a single Pod. The Pod MUST be verifiy
+    that it is running. The RS MUST scale to two replican and verify the scale count
```
Comment (dims) [inline test/conformance/testdata/conformance.yaml:710]: ugh, we missed a typo ... `replican` -> `replicas`

File: test/conformance/testdata/conformance.yaml [hunk 2]
```diff
@@ -704,6 +704,13 @@
     the Job MUST execute to completion.
   release: v1.16
   file: test/e2e/apps/job.go
+- testname: ReplicaSet, is created, Replaced and Patched
+  codename: '[sig-apps] ReplicaSet Replace and Patch tests [Conformance]'
+  description: Create a ReplicaSet (RS) with a single Pod. The Pod MUST be verifiy
```
Comment (dims) [inline test/conformance/testdata/conformance.yaml:709]: `verifiy` -> `verify`

File: test/conformance/testdata/conformance.yaml (hunk 1 above)
Comment (Riaankl) [reply test/conformance/testdata/conformance.yaml:710]: Very sharp eye Dims!
I'll fix it quickly.

File: test/conformance/testdata/conformance.yaml (hunk 2 above)
Comment (Riaankl) [reply test/conformance/testdata/conformance.yaml]: I should not have missed that.
Thanks Dims. Updated

File: test/conformance/testdata/conformance.yaml (hunk 1 above)
Comment (Riaankl) [reply test/conformance/testdata/conformance.yaml]: Updated, Thank Dims

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-kind-ipv6

Unrelated flake
```
• Failure [81.274 seconds]
[sig-node] Probing container
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/node/framework.go:23
  should be ready immediately after startupProbe succeeds [It]
  /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/node/container_probe.go:376
skipped 16435 lines unfold_more
[Fail] [sig-node] Probing container [It] should be ready immediately after startupProbe succeeds 
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/leafnodes/runner.go:113
```

**Comment by spiffxp** [comment]
/milestone v1.21
Will look later, but this is subject to Test Freeze, not Code Freeze

**Comment by dims** [comment]
Test grid is nice and clean (and green). passes all the [requirements](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements) AFAICT.

/approve
/lgtm

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**